config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundleTest.java

```diff
@@ -0,0 +1,44 @@
+package com.ca.apim.gateway.cagatewayconfig.bundle.builder;
+
+import com.ca.apim.gateway.cagatewayconfig.ProjectInfo;
+import com.ca.apim.gateway.cagatewayconfig.beans.Encass;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+public class AnnotatedBundleTest {
+
+    @Test
+    void testApplyUniqueName() {
+        String uniqueSeparator = "::";
+        String policeName = "TestPolicy";
+        Encass encass = new Encass();
+        AnnotatedEntity<Encass> encassAnnotatedEntity = new AnnotatedEntity<>(encass);
+        encassAnnotatedEntity.setEntityName("TestEncass");
+        ProjectInfo projectInfo = new ProjectInfo("TestName", "TestGroup", "unspecified");
+        AnnotatedBundle annotatedBundle = new AnnotatedBundle(null, encassAnnotatedEntity, projectInfo);
+        String uniquePolicyName = annotatedBundle.applyUniqueName(policeName);
+        Assert.assertEquals(uniqueSeparator + projectInfo.getGroupName() + "."
```

Comment:
[optional] Unit tests need to be as simple as possible. Especially, one needs to easily understand what is being tested in the sense, what are inputs and outputs.

Here, I prefer Assert.assertEquals("::TestGroup::TestName::TestEncass", ...